### PR TITLE
[41698] Global search: Text input dissapears if you tab to the search icon

### PR DESCRIPTION
**Pull request by bsatarnejad**
Bind a method to the close event and move what was written in the one-line statement to the function. When assigning currentValue to select.searchTerm in the html file, searchTerm value was not set correctly (It is null). In the onClose method, by assigning currentvalue to  the searchTerm property of ng-select instance, It works correctly. select and ng-select instance are different?

https://community.openproject.org/projects/openproject/work_packages/41698/activity

### CODE REVIEW

File: frontend/src/app/core/global_search/input/global-search-input.component.ts [hunk 1]
```diff
@@ -57,6 +57,11 @@ import { ApiV3Service } from '../../apiv3/api-v3.service';
 
 export const globalSearchSelector = 'global-search-input';
 
+interface SearchResultItems {
+  items:[];
```
Comment (HDinger) [inline frontend/src/app/core/global_search/input/global-search-input.component.ts]: This is basically an array of type `any`. I assume you want it to be of type ` SearchResultItem|SearchOptionItem`?

Comment (HDinger) [inline frontend/src/app/core/global_search/input/global-search-input.component.ts]: This is basically an array of type `any`. I assume you want it to be of type ` SearchResultItem`?

File: frontend/src/app/core/global_search/input/global-search-input.component.ts [hunk 2]
```diff
@@ -217,8 +223,13 @@ export class GlobalSearchInputComponent implements AfterViewInit, OnDestroy {
...
   }
 
+  public onClose() {
+    this.ngSelectComponent.ngSelectInstance.searchTerm = this.currentValue;
+  }
+
   public clearSearch() {
-    this.currentValue = this.ngSelectComponent.ngSelectInstance.searchTerm = '';
+    this.currentValue = '';
+    this.ngSelectComponent.ngSelectInstance.searchTerm = '';
```
Comment (HDinger) [inline frontend/src/app/core/global_search/input/global-search-input.component.ts]: This part (`this.ngSelectComponent.ngSelectInstance.searchTerm`) is called quite often in the file. Please extract it to a method, ideally a setter.

Comment (HDinger) [inline frontend/src/app/core/global_search/input/global-search-input.component.ts:232]: This part (`this.ngSelectComponent.ngSelectInstance.searchTerm`) is called quite often in the file. Please extract it to a method, ideally a setter.

**Review by HDinger — CHANGES_REQUESTED** [review]
It seems like the search bar now never closes when I actually search.
search_bar. I guess when I search, I expect it to be closed afterwards.

**Comment by bsatarnejad** [comment]
> It seems like the search bar now never closes when I actually search. search_bar. I guess when I search, I expect it to be closed afterwards.

Hi @HDinger I tested it, this behavior has existed, it's not related to the changes on this PR. I agree, it would be better to close the drop-down after search. I will fix it.

File: frontend/src/app/core/global_search/input/global-search-input.component.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
   >
     <i class="icon-arrow-left1" aria-hidden="true"></i>
   </button>
-
+
```
Comment (HDinger) [inline frontend/src/app/core/global_search/input/global-search-input.component.html]: Unnecessary spacing added here..

File: frontend/src/app/core/global_search/input/global-search-input.component.ts [hunk 4]
```diff
@@ -57,6 +57,11 @@ import { ApiV3Service } from '../../apiv3/api-v3.service';
 
 export const globalSearchSelector = 'global-search-input';
 
+interface SearchResultItems {
+  items:SearchResultItem|SearchOptionItem[];
```
Comment (HDinger) [inline frontend/src/app/core/global_search/input/global-search-input.component.ts]: Should both options be an array? So like this `SearchResultItem[]|SearchOptionItem[]`? Or am I missing something here?

**Review by HDinger — COMMENTED** [review]
Looking good so far, I only have minor remarks.

File: frontend/src/app/core/global_search/input/global-search-input.component.ts (hunk 4 above)
Comment (bsatarnejad) [reply frontend/src/app/core/global_search/input/global-search-input.component.ts]: Hi @HDinger yes both should be array. I had written it wrongly.

**Review by HDinger — APPROVED** [review]

**Comment by HDinger** [comment]
Merging as the failing spec is unrelated.

**Pull request merged**